### ISSUE TITLE
Reintroduce input field masking for form object component

### ISSUE THREAD

**Issue by elvece** [Start9Labs/embassy-os#1370]
Reintroduce masking feature for text inputs that utilize the `form-object component`

Find a comprehensive solution using reactive angular forms that can properly track masked/unmasked data in the recursive form object component structure. This is mainly used for service configurations.

[This](https://github.com/Start9Labs/embassy-os/blob/be11b786ee7288a971a8fc04f682c98904931684/frontend/projects/ui/src/app/components/form-object/form-object.component.html#L82) is what was previously removed - the reason for is explained below.

**Comment by ProofOfKeags** [comment]
Why were they removed to begin with?

**Comment by elvece** [comment]
They were removed to resolve [this issue ](https://github.com/Start9Labs/embassy-os/issues/918), which @MattDHill and I decided was a more painful and dangerous UX (as the pw manager could override the already saved pw for EOS login if saved for these other input fields).

There was not a straightforward solution due to data flow ordering in angular bindings and change events; however, bluj and I did devise a solution in the basic case. The dynamically generated config form objects present a challenge with how the two way bound model is preserved, so that case still needs to be evaluated and is mainly what this ticket is for. 

In an effort to streamline the 0.3.0.1 release, removing them was the MVP.
